File: app/src/main/java/com/example/moneymanager/TransactionAdapter.java
package com.example.moneymanager;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

public class TransactionAdapter extends RecyclerView.Adapter<TransactionAdapter.ViewHolder> {
    private ArrayList <TransactionList> mTransactionList;

    public TransactionAdapter(ArrayList<TransactionList> transactionList) {
        this.mTransactionList = transactionList;
    }


    public class ViewHolder extends RecyclerView.ViewHolder {
        public ImageView transactionImages;
        public TextView transactionNames, transactionDates, transactionBalance;


        public ViewHolder(@NonNull View itemView) {
            super(itemView);

            transactionImages = itemView.findViewById(R.id.ivTransactionLogo);
            transactionNames = itemView.findViewById(R.id.tvNames);
            transactionDates = itemView.findViewById(R.id.tvDates);
            transactionBalance = itemView.findViewById(R.id.tvBalance);
        }
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        Context context = parent.getContext();
        LayoutInflater inflater = LayoutInflater.from(context);

        // Inflate the custom layout
        View transactionView = inflater.inflate(R.layout.rv_transaction_items,
                parent, false);

        // Return a new holder instance
        TransactionAdapter.ViewHolder viewHolder = new ViewHolder(transactionView);
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        TransactionList transactionList = mTransactionList.get(position);
        holder.transactionImages.setImageResource(transactionList.getTransactionLogos());
        if (position == 1) {
            holder.transactionImages.setPadding(5,5,5,5);
        }
        holder.transactionNames.setText(transactionList.getTransactionNames());
        holder.transactionDates.setText(transactionList.getTransactionDate());
        holder.transactionBalance.setText(transactionList.getBudget());

    }

    @Override
    public int getItemCount() {
        return mTransactionList.size();
    }
}

File: app/src/main/java/com/example/moneymanager/TransactionAdapter.kt
package com.example.moneymanager

import androidx.recyclerview.widget.RecyclerView
import android.view.ViewGroup
import android.view.LayoutInflater
import com.example.moneymanager.databinding.RvTransactionItemsBinding
import com.example.moneymanager.model.Transaction

class TransactionAdapter(private val transactions: List<Transaction>) : RecyclerView.Adapter<TransactionAdapter.ViewHolder>() {
    inner class ViewHolder(val transactionItemsBinding:RvTransactionItemsBinding) : RecyclerView.ViewHolder(transactionItemsBinding.root) {
//        val transactionImage: ImageView = itemView.findViewById(R.id.ivTransactionLogo)
//        val transactionName: TextView = itemView.findViewById(R.id.tvNames)
//        val transactionDate: TextView = itemView.findViewById(R.id.tvDates)
//        val transactionBalance: TextView = itemView.findViewById(R.id.tvBalance)

    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        // Inflate the custom layout
        val binding = RvTransactionItemsBinding.inflate(LayoutInflater.from(parent.context),
        parent, false)
        // Return a new holder instance
        return ViewHolder(binding)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        with(holder) {
            with(transactions[position]) {
                transactionItemsBinding.ivTransactionLogo.setImageResource(logo)
                if (position == 1) {
                    transactionItemsBinding.ivTransactionLogo.setPadding(5,5,5,5)
                }
                transactionItemsBinding.tvNames.text = this.name
                transactionItemsBinding.tvDates.text = this.date
                transactionItemsBinding.tvBalance.text = this.budget
            }
        }
    }

    override fun getItemCount(): Int {
        return transactions.size
    }
}

